MSBLOC.Core/Services/Parser.cs

```diff
@@ -21,31 +74,30 @@ public Parser(ILogger<Parser> logger = null)
             Logger = logger ?? new NullLogger<Parser>();
         }
 
-        public ParsedBinaryLog Parse(string resourcePath)
+        public ParsedBinaryLog Parse(string resourcePath, string cloneRoot = "")
         {
             var warnings = new List<BuildWarningEventArgs>();
             var errors = new List<BuildErrorEventArgs>();
             var binLogReader = new StructuredLogger::Microsoft.Build.Logging.BinaryLogReplayEventSource();
 
             var readRecords = binLogReader.ReadRecords(resourcePath).ToList();
 
-            var projectFileLookup = new Dictionary<string, Dictionary<string, string>>();
+            var solutionDetails = new SolutionDetails(cloneRoot);
 
             foreach (var record in readRecords)
```

Comment:
Probably can be a Parallel.ForEach as the iterations are independant. Just need to make sure the underlying Dictionary implementation is a ConcurrentDictionary.

Reply:
I actually do need these events to be read in order. `ProjectStartedEventArgs` will come before `BuildWarningEventArgs` and `BuildErrorEventArgs`.

I did not mean for `ToList` to be called though. That was from a debugging session.